### ISSUE TITLE
Unable to build MR updates

### ISSUE THREAD

**Issue by hho** [jenkinsci/gitlab-plugin#227]
I have my Jenkins job set up like ramielrowe outlined [here](https://github.com/jenkinsci/gitlab-plugin/issues/172#issuecomment-178654421).

This worked with 1.1.28.
Since 1.1.31 I only get IAEs when attempting to build anything:

```
Building in workspace /var/lib/jenkins/jobs/hho/jobs/dummy-merge-requests/workspace
FATAL: Null value not allowed as an environment variable: gitlabUserEmail
java.lang.IllegalArgumentException: Null value not allowed as an environment variable: gitlabUserEmail
    at hudson.EnvVars.put(EnvVars.java:356)
    at hudson.model.StringParameterValue.buildEnvironment(StringParameterValue.java:56)
    at hudson.model.ParametersAction.buildEnvVars(ParametersAction.java:89)
    at hudson.model.AbstractBuild.getEnvironment(AbstractBuild.java:950)
    at hudson.maven.AbstractMavenBuild.getEnvironment(AbstractMavenBuild.java:56)
    at hudson.maven.MavenModuleSetBuild.getEnvironment(MavenModuleSetBuild.java:166)
    at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1046)
    at hudson.scm.SCM.checkout(SCM.java:485)
    at hudson.model.AbstractProject.checkout(AbstractProject.java:1269)
    at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:607)
    at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
    at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:529)
    at hudson.model.Run.execute(Run.java:1738)
    at hudson.maven.MavenModuleSetBuild.run(MavenModuleSetBuild.java:531)
    at hudson.model.ResourceController.execute(ResourceController.java:98)
    at hudson.model.Executor.run(Executor.java:410)
Finished: FAILURE
```

The gitlabUserEmail parameter never gets filled, even though there is a `user_email` property in the push web hooks JSON.

Also, even though the job only gets triggered with a "push" web hook ("Rebuild MR on push to source" is active), the gitlabActionType parameter gets set to "MERGE".

(Jenkins 1.652, Gitlab-Plugin 1.1.31, GitLab 8.5.4)

**Comment by avdv** [comment]
Same problem here after updating the plugin from 1.1.28 to 1.1.31.

Jenkins: 1.652
Gitlab: 8.4.2-ee

**Comment by NITEMAN** [comment]
Same here.

Jenkins: 1.652
Gitlab: 8.5.4-ce

**Comment by abcfy2** [comment]
Same here.

Jenkins: 1.652
Gitlab: 8.5.4-ce

**Comment by avdv** [comment]
Apparently, this problem is already known since 2nd January, but the new version had been released regardless. :open_mouth: 

See #170 and #216.

**Comment by ZeroPointEnergy** [comment]
Is there a workaround for the problem until this is released?

**Comment by avdv** [comment]
> Is there a workaround for the problem until this is released?

Check out the master branch and run `mvn package`. Then, install the generated `gitlab-plugin-1.1.32-SNAPSHOT.hpi` file from the `target` directory.

**Comment by omehegan** [comment]
This was fixed in version 1.1.32. See https://github.com/jenkinsci/gitlab-plugin/blob/master/CHANGELOG.md. Please reopen if not fixed.
